### PR TITLE
Improvement: DO-2039 Graph components docs improvements

### PR DESCRIPTION
**Pull request by tamara-causalens**
## Motivation and Context
Small docs improvements:
- Adds an example on how to use EditorMode in CausalGraphViewer
- Fixes the import for EdgeConstraints in VisualEdgeEncoder

## PR Checklist:

- [x] I have implemented all requirements? (see JIRA, project documentation).
- [x] I am not affecting someone else's work, If I am, they are included as a reviewer.
- [ ] I have added relevant tests (unit, integration or regression).
- [ ] I have added comments to all the bits that are hard to follow.
- [x] I have added/updated Documentation.
- [ ] I have updated the appropriate changelog with a line for my changes.

### CODE REVIEW

File: packages/dara-components/dara/components/graphs/components/causal_graph_viewer.py [hunk 1]
```diff
@@ -52,7 +52,24 @@ class CausalGraphViewer(BaseGraphComponent):
     The causal graph can be edited by setting `editable=True`. The editor mode can be set to `EditorMode.DEFAULT`
     (default), `EditorMode.PAG` or `EditorMode.RESOLVER`. The `EditorMode.DEFAULT` mode assumes all edges
     are directed (the causal graph is a DAG). The `EditorMode.PAG` mode displays all edge types (beyond directed), while
-    `EditorMode.RESOLVER` allows users to confirm and accept edges.
+    `EditorMode.RESOLVER` allows users to confirm and accept edges. To change the editor mode, set the `editor_mode`
+    parameter.
+
+    ```python
+    from dara.components.graphs import CausalGraphViewer
+    from dara.components.graphs.definitions import EditorMode
```
Comment (patricia-causalens) [inline packages/dara-components/dara/components/graphs/components/causal_graph_viewer.py:60]: You can import them from the top level `from dara.components import EditorMode, CausalGraphViewer` makes it a bit easier for the example

File: packages/dara-components/dara/components/graphs/components/edge_encoder.py [hunk 2]
```diff
@@ -89,7 +89,8 @@ class VisualEdgeEncoder(StyledComponentInstance):
     ```python
     from cai_causal_graph import CausalGraph
     from cai_causal_graph.type_definitions import EdgeConstraint as EdgeConstraintType
-    from dara.components.graphs import VisualEdgeEncoder, EdgeConstraint
+    from dara.components.graphs import VisualEdgeEncoder
+    from dara.components.graphs.components.edge_encoder import EdgeConstraint
```
Comment (patricia-causalens) [inline packages/dara-components/dara/components/graphs/components/edge_encoder.py:93]: Did this not work for you? Tried locally and we seem to be able to do as before or `from dara.components import VisualEdgeEncoder, EdgeConstraint`

**Review by patricia-causalens — APPROVED** [review]
Looks good, just a couple of suggestions to make the imports shorter.

**Review by krzysztof-causalens — APPROVED** [review]

File: packages/dara-components/dara/components/graphs/components/causal_graph_viewer.py (hunk 1 above)
Comment (tamara-causalens) [reply packages/dara-components/dara/components/graphs/components/causal_graph_viewer.py]: adapted across the docstrings - thanks!

File: packages/dara-components/dara/components/graphs/components/edge_encoder.py (hunk 2 above)
Comment (tamara-causalens) [reply packages/dara-components/dara/components/graphs/components/edge_encoder.py]: There was feedback that it didn't work (might be due to older version). Tried it and it does work as expected so I added it in the example as well

File: packages/dara-components/dara/components/graphs/components/causal_graph_viewer.py [hunk 3]
```diff
@@ -67,7 +83,7 @@ class CausalGraphViewer(BaseGraphComponent):
         offer any protection against edge crossings.
 
     ```python
-    from dara.components.graphs import CausalGraphViewer
+    from dara.components import CausalGraphViewer
     from dara.components.graphs.graph_layout import PlanarLayout
```
Comment (patricia-causalens) [inline packages/dara-components/dara/components/graphs/components/causal_graph_viewer.py:87]: You can also import the layouts from top level :)

**Review by patricia-causalens — APPROVED** [review]
LGTM, just a last minor comment :)

File: packages/dara-components/dara/components/graphs/components/causal_graph_viewer.py (hunk 3 above)
Comment (tamara-causalens) [reply packages/dara-components/dara/components/graphs/components/causal_graph_viewer.py]: great call! done

**Pull request merged**